test/integration/targets/azure_rm_virtualmachine_scaleset/tasks/main.yml

```diff
@@ -50,14 +55,33 @@
         managed_disk_type: Standard_LRS
   register: results
 
-- name: Assert that VMSS ran
+- name: Assert that VMSS was created
+  assert:
+    that: results.changed
+
+- name: Retrieve scaleset facts
+  azure_rm_virtualmachine_scaleset_facts:
+    resource_group: "{{ resource_group }}"
+    name: testVMSS{{ rpfx }}
+    format: curated
+  register: output_scaleset
+
+- name: Modify scaleset (set upgradePolicy to Automatic and capacity to 3)
+  set_fact:
+    body: "{{ output_scaleset.ansible_facts.azure_vmss[0] | combine({'capacity': 1}, recursive=True) }}"
+
+- name: Update VMSS
```

Comment:
I would get rid of this, this format is meant to be removed in this version, this is the deprecation warning you get

```
[DEPRECATION WARNING]: Using variables for task params is unsafe, especially if the variables come from an external source like facts. This 
feature will be removed in version 2.6. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```

Reply:
ok, i will do this. but actually i wonder how this could be easily tested.
even if using variables for task params is unsafe, someone should be able to use facts to get existing resource params and create task out of that manually.

Do you think there should be some mechanism for updating existing resources?
I am thinking of a scenario where:
- user has existing resource created somehow (for example using terraform)
- they want to update single param only without creating entire definition

Does this "using variables for task params" apply just for top level, or or entire subtree?